src/backend/dx12/src/command.rs

```diff
@@ -1196,24 +1203,50 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
                 }
             })
             .collect();
+        
+        // Add items to the viewport cache if necessary
+        if self.viewport_cache.len() < first_viewport as usize + viewports.len() {
+            self.viewport_cache.push(
+                d3d12::D3D12_VIEWPORT { 
+                    TopLeftX: 0.,
+                    TopLeftY: 0.,
+                    Width: 0.,
+                    Height: 0.,
+                    MinDepth: 0.,
+                    MaxDepth: 0.,
+                }
+            );
+        }
+        for (viewports_index, cache_index) in ((first_viewport as usize)..(viewports.len() + first_viewport as usize)).enumerate() {
```

Comment:
nit: could be merged with the upper iteratoration.
Would be nice to simplify this to `for (i, viewport) in viewports.enumerate() { .. }`.